avm/src/vm/memory.rs

```diff
@@ -661,30 +397,23 @@ impl HandlerMemory {
     loop {
       if i < s_bytes.len() {
         let s_slice = &s_bytes[i..i+8];
-        s_mem.push(i64::from_ne_bytes(s_slice.try_into().unwrap()));
+        s_mem.push((std::usize::MAX, i64::from_ne_bytes(s_slice.try_into().unwrap())));
         i = i + 8;
       } else {
         break
       }
     }
-    let mem_size = s_mem.len();
-    HandlerMemory {
-      gmem: &Program::global().gmem,
-      mem: s_mem,
-      fractal_mem: vec![],
-      either_mem: vec![-1; mem_size],
-      registers_ish: vec![0; mem_size],
-      is_fixed: false,
-      closure_args: Vec::new(),
-    }
+    s_mem
   }
 
-  pub fn hm_to_string(self: &HandlerMemory) -> String {
-    let s_len = self.mem[0] as usize;
-    unsafe {
-      let s_u8 = slice::from_raw_parts(self.mem[1..].as_ptr().cast::<u8>(), s_len*8);
-      let s = str::from_utf8(&s_u8[0..s_len]).unwrap();
-      s.to_string()
+  pub fn fractal_to_string(f: &[(usize, i64)]) -> String {
+    let s_len = f[0].1 as usize;
+    let mut s_bytes: Vec<u8> = Vec::new();
+    for i in 1..f.len() {
+      let mut b = f[i].1.clone().to_ne_bytes().to_vec();
```

Comment:
hmm might be missing something, but [`to_vec()`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.to_vec) is effectively a clone. Can we get away without the additional `clone()` or does the borrow checker complain? If so, can we leave a comment about it

Reply:
So I did get problems here, but looking at this I think I can simplify it. `to_vec` was required to solve it because it has some sort of "copy-on-write"-like behavior where the basic `clone` doesn't work (because it's just a pointer and cloning a pointer still points at the same underlying data).

This is probably one of the functions I finished up around midnight and once it compiled and worked correctly I basically ignored since.